src/include/cell_inline.h

```diff
@@ -1082,8 +1084,23 @@ __cell_unpack_window_cleanup(WT_SESSION_IMPL *session, const WT_PAGE_HEADER *dsk
      * No delete              txnid=MAX, ts=MAX,            txnid=MAX, ts=MAX,
      *                        durable_ts=NONE               durable_ts=NONE
      */
+
+    if (WT_READING_CHECKPOINT(session) && session->checkpoint_write_gen != 0) {
+        /*
+         * When reading a checkpoint, override the tree's base write generation with the write
+         * generation from the global metadata, which might be newer. This comes into play if the
+         * tree checkpoint is from an older database run than the global checkpoint, which can
+         * happen if checkpointing skips the tree at the right points. Bypass this logic if the
+         * checkpoint write generation isn't set because the checkpoint is from an older version of
+         * WiredTiger; in that case we use the tree's write generation and hope for the best.
+         */
+        write_gen = session->checkpoint_write_gen;
+        WT_ASSERT(session, write_gen >= S2BT(session)->base_write_gen);
```

Comment:
I have a doubt about this condition. In the following scenario,
1. connection base write gen is 10, so 10 is stored in the metadata as part of the checkpoint.
2. Restarted and connection base write gen is incremented to 20 and it is saved in the btree base write gen.
3. Reading it from the checkpoint can cause problems?

Reply:
It isn't saved in the btree base write gen if it's a checkpoint. Recall this change from the original branch: [4570ff5](https://github.com/wiredtiger/wiredtiger/pull/7661/commits/4570ff593aa766c0533bf2ba6f972307bb495914); that makes the btree base write gen for checkpoint btrees always be the write gen from the tree-level checkpoint. The new problem last week was that the tree-level checkpoint can be older than the global checkpoint, and if so we want to use the global checkpoint's write generation instead. But this should only move the write generation forward; hence the assertion here.

Reply:
Thanks, I forgot the change in bt_handle.c.